src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -498,6 +498,16 @@ CustomTextLayout::CustomTextLayout(IDWriteFactory2* const factory,
             glyphRunDescription.stringLength = run.textLength;
             glyphRunDescription.textPosition = run.textStart;
 
+            // Calculate the position *after* drawing, this will be used if we're in RTL.

```

Comment:
The comment explains what it does, but not why. Please include here and/or below that the reason WHY we precalculate the final position is because DirectX expects the origin to the RIGHT of the text for RTL and LEFT for LTR.

Reply:
Okay, redid the commenting.